docker/release/nanoserver/Dockerfile

```diff
@@ -1,48 +1,44 @@
 
-# Setup PowerShell - Log-to > C:\Docker.log
-SHELL ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe", "-command"]
-ADD $POWERSHELL_ZIP /powershell-win10-x64.zip
-ENV POWERSHELL_SHA256=$POWERSHELL_SHA256
+# Prepare shell environment to Log-to > C:\Dockerfile.log
+SHELL ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe","-command","$ErrorActionPreference = 'Stop';","$ConfirmPreference = 'None';",`
+    "$VerbosePreference = 'Continue';","Start-Transcript -path C:\\Dockerfile.log -append -IncludeInvocationHeader;","$PSVersionTable|Write-Output;",`
+    "$WinVer = $(Get-ItemProperty -Path 'HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion');",`
+    "$OsVer = $('OperatingSystem:  '+ $WinVer.ProductName +' '+$WinVer.EditionId+' '+$WinVer.InstallationType);",`
+    "$BldVer = $('FullBuildString:  '+ $WinVer.BuildLabEx); Write-Output -InputObject $OsVer,$BldVer; "]
+WORKDIR /
 
 # Install PowerShell package and clean up
-RUN $ErrorActionPreference='Stop'; `
-    $ConfirmPreference='None'; `
-    $VerbosePreference='Continue'; `
-    Start-Transcript -path C:\Dockerfile.log -append -IncludeInvocationHeader ; `
-    $PSVersionTable | Write-Output ; `
-    $VerInfo = Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion' ; `
-    ('FullBuildString: '+$VerInfo.BuildLabEx) | Write-Output ; `
-    ('OperatingSystem:  '+$VerInfo.ProductName+' '+$VerInfo.EditionId+' '+$VerInfo.InstallationType)  | Write-Output ; `
-    [System.IO.FileInfo]$ZipFile = Get-Item -Path ./powershell-win10-x64.zip ; `
-    [String]$ZipHash=(Get-FileHash  -Path $ZipFile -Algorithm SHA256 | select -ExpandProperty Hash) ; `
-    If ($ZipHash -notmatch $Env:POWERSHELL_SHA256) { `
-      Throw [String]$('['+$ZipHash+'] does not match ['+$Env:POWERSHELL_SHA256+']!') `
-    } ; `
-    New-Item -Path $Env:ProgramFiles/PowerShell -ItemType Directory -Force | out-null ; `
-    [System.IO.DirectoryInfo]$PsFolder=New-Item -Path $Env:ProgramFiles\PowerShell -ItemType Directory -Force ; `
-    Add-Type -AssemblyName System.IO.Compression.ZipFile ; `
-    [System.IO.Compression.ZipFile]::ExtractToDirectory($ZipFile,$PsFolder) ; `
-    if (Get-ChildItem -Path $PsFolder/powershell.exe) { `
-      Remove-Item -Path $ZipFile ; `
-      New-Item -Type SymbolicLink -Path $PsFolder\ -Name Core -Value $PsFolder `
-    } else { throw 'Installation failed!  See c:\Dockerfile.log' } ;
+ADD $POWERSHELL_ZIP /PowerShell-win10-win2k16-x64.zip
+RUN [System.IO.FileInfo]$ZipFile = Get-Item -Path ./powershell-win10-win2k16-x64.zip;`
+    [String]$ZipHash = (Get-FileHash  -Path $ZipFile -Algorithm SHA256 | select -ExpandProperty Hash);`
+    If ($ZipHash -notmatch $Env:POWERSHELL_SHA256){`
+        Throw [String]$('['+$ZipHash+'] does not match ['+$Env:POWERSHELL_SHA256+']!')`
+    }`
+    [System.IO.DirectoryInfo]$PsFolder=New-Item -Path $Env:ProgramFiles\PowerShell -ItemType Directory -Force;`
```

Comment:
Expand-Archive would simplify this. [Here](https://github.com/PowerShell/PowerShell/blob/master/docs/installation/windows.md#online-deployment-of-powershell-core) is an example.